### PR TITLE
CBG-2698: [2.8.4 backport] conflictIncludesRev support for proposeChanges (#6050)Co-authored-by: Adam Fraser <adam.fraser@couchbase.com>

### PR DESCRIPTION
**Pull request by gregns1**
CBG-2698

Backport the conflictIncludesRev support for propseChanges. NOTE: this PR has dependency changes and an upgrade to go version in it so PR against master 2.8 manifest file needs to be merged first. 

## Pre-review checklist
- [ ] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [ ] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [ ] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

**Comment by adamcfraser** [comment]
@gregns1 Is there a reason the dependency/go-version changes are getting pulled into this PR?

**Comment by gregns1** [comment]
@adamcfraser  I should be separating these into separate PRs. Not sure why I thought I should get it done in one when this back port doesn't rely on these dependency changes, they are part of some of the other back port tickets. Sorry I'll get that done over the weekend and move the dependencies into another PR. The PR against master 2.8 manifest is the main one for the dependency changes. The upgrade to compress needed to be done with go upgrade as the old version of go didn't work with that version of compress.

**Comment by bbrks** [comment]
Needs rebasing on latest `release/2.8.4` to pick up gofmt, build fixes, etc.

File: rest/utilities_testing.go [hunk 1]
```diff
@@ -674,6 +674,9 @@ type BlipTesterSpec struct {
 	// Run Sync Gateway in "No conflicts" mode.  Will be propgated to the underyling RestTester
 	noConflictsMode bool
 
+	// If an underlying RestTester is created, it will propagate this setting to the underlying RestTester.
+	guestEnabled bool
```
Comment (adamcfraser) [inline rest/utilities_testing.go]: This doesn't look like it's actually wired up to anything on this branch - can it just be omitted from the test?

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: rest/utilities_testing.go (hunk 1 above)
Comment (gregns1) [reply rest/utilities_testing.go]: So this was added as `TestProposedChangesIncludeConflictingRev` uses guestEnabled in the blip tester spec on this branch

Comment (bbrks) [reply rest/utilities_testing.go]: It sets it but nothing reads this value. That means either:
- There is missing functionality inside BlipTester that needs to be pulled in as well
- Or the test setting something it doesn't need to for it to run properly

Comment (gregns1) [reply rest/utilities_testing.go]: Ah I see good point. I've just removed it and doesn't effect the test when I've run it against walrus and server so will push those changes up

**Review by bbrks — APPROVED** [review]

**Pull request merged**